Change downgrade process (downgrade bootstrap first)

TeddyAndrieux opened the pull request:
**Component**:

'salt', 'scripts', 'downgrade'
<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 

Downgrade process is wrong, some ext_pillar need a salt-master in the same version to be computed

**Summary**:

- Downgrade bootstrap stuff before calling the downgrade orchestrate
- No longer downgrade etcd at the same time of the node

---

Fixes: #2101

TeddyAndrieux committed:
script,salt: Downgrade `bootstrap` before other nodes

Some ext_pillar need a salt-master already in the dest_version to be
computed, so we need to first downgrade salt-master before applying
highstate on other nodes

Fixes: #2101

TeddyAndrieux committed:
salt: Skip `etcd` role when calling highstate downgrade

When downgrading we need to first downgrade all the nodes and after take
care of the etcd cluster, that why we need to skip `etcd` role when
calling highstate on nodes

bert-e commented:
# Hello teddyandrieux,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/GH-2101-change-downgrade-process` into
`development/2.0`
* [w/2.1/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.1` 
* [w/2.2/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.2` 
* [w/2.3/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.3...w/2.3/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.3` 
* [w/2.4/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.4...w/2.4/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.4` 
* [w/2.5/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.5` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

TeddyAndrieux committed:
salt: Add a `refresh_pillar` in upgrade and downgrade orchestrate

When deploying node we need a pillar update to date, we already have a
function checking for pillar value in deploy_node, but since the
`refresh_pillar` is always async we first do a refresh everywhere before
calling any deploy_node orchestrate

bert-e commented:
# History mismatch



Merge commit #f4498507c8795bb89ec58e4513f5cb4b94c6601e on the integration branch
`w/2.1/bugfix/GH-2101-change-downgrade-process` is merging a branch which is neither the current
branch `bugfix/GH-2101-change-downgrade-process` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `bugfix/GH-2101-change-downgrade-process` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

TeddyAndrieux commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/GH-2101-change-downgrade-process` into
`development/2.0`
* [w/2.1/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.1` 
* [w/2.2/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.2` 
* [w/2.3/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.3...w/2.3/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.3` 
* [w/2.4/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.4...w/2.4/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.4` 
* [w/2.5/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.5` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

bert-e commented:
# History mismatch



Merge commit #ae3d85c0858593f86c1ce0cbc8afbe0d01b65e12 on the integration branch
`w/2.1/bugfix/GH-2101-change-downgrade-process` is merging a branch which is neither the current
branch `bugfix/GH-2101-change-downgrade-process` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `bugfix/GH-2101-change-downgrade-process` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

TeddyAndrieux commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/GH-2101-change-downgrade-process` into
`development/2.0`
* [w/2.1/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.1` 
* [w/2.2/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.2` 
* [w/2.3/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.3...w/2.3/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.3` 
* [w/2.4/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.4...w/2.4/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.4` 
* [w/2.5/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.5` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

Ebaneck commented on salt/metalk8s/orchestrate/downgrade/init.sls:
```diff
@@ -10,6 +10,20 @@ Execute the downgrade prechecks:
 
+# In salt 2018.3 we can not do synchronous pillar refresh, so add a sleep
+# see https://github.com/saltstack/salt/issues/20590
+Refresh pillar everywhere and wait a bit:
```
Minor: ```Wait for pillar refresh to complete``` ??

TeddyAndrieux replied:
True it's better, I will change

Ebaneck reviewed (changes requested):
Slight grammar check

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @ebaneck

Ebaneck commented on salt/metalk8s/orchestrate/downgrade/init.sls:
```diff
@@ -10,6 +10,20 @@ Execute the downgrade prechecks:
+    - test.sleep:
+      - length: 20
+    - require:
+      - salt: Refresh pillar everywhere and wait a bit
```
ditto

Ebaneck commented on salt/metalk8s/orchestrate/downgrade/init.sls:
```diff
@@ -39,7 +48,7 @@ Wait for API server to be available on {{ node }}:
   - verify_ssl: false
   - require:
-    - salt: Execute the downgrade prechecks
+    - module: Refresh pillar everywhere and wait a bit
```
ditto

Ebaneck commented on salt/metalk8s/orchestrate/upgrade/init.sls:
```diff
@@ -22,6 +22,20 @@ Upgrade etcd cluster:
 
+# In salt 2018.3 we can not do synchronous pillar refresh, so add a sleep
+# see https://github.com/saltstack/salt/issues/20590
+Refresh pillar everywhere and wait a bit:
```
ditto

Ebaneck commented on salt/metalk8s/orchestrate/upgrade/init.sls:
```diff
@@ -22,6 +22,20 @@ Upgrade etcd cluster:
+    - test.sleep:
+      - length: 20
+    - require:
+      - salt: Refresh pillar everywhere and wait a bit
```
ditto

Ebaneck commented on salt/metalk8s/orchestrate/upgrade/init.sls:
```diff
@@ -46,7 +60,7 @@ Wait for API server to be available on {{ node }}:
   - verify_ssl: false
   - require:
-    - salt: Upgrade etcd cluster
+    - module: Refresh pillar everywhere and wait a bit
```
ditto

bert-e commented:
# History mismatch



Merge commit #4333b4b5883fe8cc52e462c89676ee93dc41322a on the integration branch
`w/2.1/bugfix/GH-2101-change-downgrade-process` is merging a branch which is neither the current
branch `bugfix/GH-2101-change-downgrade-process` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `bugfix/GH-2101-change-downgrade-process` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

TeddyAndrieux commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/GH-2101-change-downgrade-process` into
`development/2.0`
* [w/2.1/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.1` 
* [w/2.2/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.2` 
* [w/2.3/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.3...w/2.3/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.3` 
* [w/2.4/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.4...w/2.4/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.4` 
* [w/2.5/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.5` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @ebaneck

gdemonet commented on salt/metalk8s/orchestrate/deploy_node.sls:
```diff
@@ -116,6 +116,13 @@ Run the highstate:
     - saltenv: metalk8s-{{ version }}
+    - pillar:
+        # Add ability to override node roles to not apply all the highstate
+        # e.g.: Skipping etcd when downgrading
```
It works, sure, but doesn't look super good IMHO... Could we use something dedicated to "skipping" roles? E.g. `pillar.highstate.skip_roles", and let the `top.sls` filter this out?

TeddyAndrieux replied:
Sure could be better i will

TeddyAndrieux replied:
Ok just thinking about it and that mean that we will not support it for current released version as we can't change the top.sls of already released version but I think it's quite ok

gdemonet commented on salt/metalk8s/orchestrate/downgrade/init.sls:
```diff
@@ -10,6 +10,20 @@ Execute the downgrade prechecks:
+      - salt: Execute the downgrade prechecks
+  module.run:
+    - test.sleep:
+      - length: 20
```
20 seconds for a pillar refresh?! Isn't that a bit too much?

TeddyAndrieux replied:
Since we do several call to apiserver that may be on another machine I think it's quite ok and it's only one time during all the upgrade/downgrade process so it's ok IMO

gdemonet replied:
OK. We should do some profiling on our various Salt orchestrations, some time (not urgent at all, but would be interesting for us).

gdemonet reviewed (dismissed):
Minor comments, otherwise looks good to me

TeddyAndrieux committed:
salt: Add ability to `skip_roles` in salt top.sls

We may want some time to run only a part of the highstate, for example
when downgrading we want to be able to apply an highstate except `etcd`
role

bert-e commented:
# History mismatch



Merge commit #4bbb422fe5e77fd85a378ee21f71f922b8300fae on the integration branch
`w/2.1/bugfix/GH-2101-change-downgrade-process` is merging a branch which is neither the current
branch `bugfix/GH-2101-change-downgrade-process` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `bugfix/GH-2101-change-downgrade-process` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

TeddyAndrieux commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/GH-2101-change-downgrade-process` into
`development/2.0`
* [w/2.1/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.1` 
* [w/2.2/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.2` 
* [w/2.3/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.3...w/2.3/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.3` 
* [w/2.4/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.4...w/2.4/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.4` 
* [w/2.5/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.5` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @ebaneck

bert-e commented:
# History mismatch



Merge commit #d14431b6c89c96039157e9b0930cbf1529aba8cb on the integration branch
`w/2.1/bugfix/GH-2101-change-downgrade-process` is merging a branch which is neither the current
branch `bugfix/GH-2101-change-downgrade-process` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `bugfix/GH-2101-change-downgrade-process` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

TeddyAndrieux commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/GH-2101-change-downgrade-process` into
`development/2.0`
* [w/2.1/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.1` 
* [w/2.2/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.2` 
* [w/2.3/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.3...w/2.3/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.3` 
* [w/2.4/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.4...w/2.4/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.4` 
* [w/2.5/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.5` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @ebaneck

bert-e commented:
# History mismatch



Merge commit #0f9914068ba9ff39028b1ac811379291e06a370d on the integration branch
`w/2.1/bugfix/GH-2101-change-downgrade-process` is merging a branch which is neither the current
branch `bugfix/GH-2101-change-downgrade-process` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `bugfix/GH-2101-change-downgrade-process` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

TeddyAndrieux commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/GH-2101-change-downgrade-process` into
`development/2.0`
* [w/2.1/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.1` 
* [w/2.2/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.2` 
* [w/2.3/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.3...w/2.3/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.3` 
* [w/2.4/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.4...w/2.4/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.4` 
* [w/2.5/bugfix/GH-2101-change-downgrade-process](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/bugfix/GH-2101-change-downgrade-process) will be merged into `development/2.5` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @ebaneck

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

gdemonet commented on salt/metalk8s/orchestrate/deploy_node.sls:
```diff
@@ -1,6 +1,8 @@
 {%- set node_name = pillar.orchestrate.node_name %}
 {%- set version = pillar.metalk8s.nodes[node_name].version %}
 
+{%- set skip_roles = pillar.metalk8s.nodes[node_name].get('skip_roles', []) %}
```
Maybe source it from `pillar.orchestrate`?

TeddyAndrieux replied:
Hmm no, it's not really intuitif that mean if someone want to skip etcd from CLI directly when running highstate he will need to change the one in `metalk8s.nodes` and if he want to skip using `deploy_node` he can only use `orchestrate` one.
Even if it's not a feature we should use really often for me it's better to have only one place to put it in the pillar

TeddyAndrieux commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`

* :heavy_check_mark: `development/2.3`

* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`

* :heavy_check_mark: `development/2.3`

* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



Please check the status of the associated issue GH-2101.

Goodbye teddyandrieux.
